xcm/xcm-builder/src/integration_tests.rs

```diff
@@ -0,0 +1,282 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use frame_support::weights::Weight;
+
+use sp_runtime::traits::AccountIdConversion;
+
+
+use polkadot_parachain::primitives::Id as ParaId;
+
+use xcm::opaque::v0::prelude::*;
+use xcm::opaque::v0::{Response};
+use xcm::v0::{MultiLocation::*, Order};
+use xcm_executor::XcmExecutor;
+
+use crate::mock;
+
+use crate::integration_mock::{
+	AccountId, Balances, BaseXcmWeight, ExistentialDeposit, kusama_like_with_balances, XcmConfig
+};
+
+pub const ALICE: AccountId = AccountId::new([0u8; 32]);
+pub const PARA_ID: u32 = 2000;
+pub const INITIAL_BALANCE: u128 = 100_000_000_000;
+
+// Construct a `BuyExecution` order.
+fn buy_execution<C>(debt: Weight) -> Order<C> {
+	use xcm::opaque::v0::prelude::*;
+	Order::BuyExecution {
+		fees: All,
+		weight: 0,
+		debt,
+		halt_on_error: false,
+		xcm: vec![],
+	}
+}
+
+/// Scenario:
+/// A parachain transfers funds on the relaychain to another parachain's account.
+///
+/// Asserts that the parachain accounts are updated as expected.
+#[test]
+fn withdraw_and_deposit_works() {
+	let para_acc: AccountId = ParaId::from(PARA_ID).into_account();
+	let balances = vec![(ALICE, INITIAL_BALANCE), (para_acc.clone(), INITIAL_BALANCE)];
+	kusama_like_with_balances(balances).execute_with(|| {
+		let other_para_id = 3000;
+		let amount =  10 * ExistentialDeposit::get();
+		let weight = 3 * BaseXcmWeight::get();
+		let r = XcmExecutor::<XcmConfig>::execute_xcm(
+			Parachain(PARA_ID).into(),
+			Xcm::WithdrawAsset {
+				assets: vec![ConcreteFungible { id: Null, amount }],
+				effects: vec![
+					buy_execution(weight),
+					Order::DepositAsset {
+						assets: vec![All],
+						dest: Parachain(other_para_id).into(),
+					},
+				],
+			},
+			weight,
+		);
+		assert_eq!(r, Outcome::Complete(weight));
```

Comment:
```suggestion
		assert!(matches!(XcmExecutor::<XcmConfig>::execute_xcm(
			Parachain(PARA_ID).into(),
			Xcm::WithdrawAsset {
				assets: vec![ConcreteFungible { id: Null, amount }],
				effects: vec![
					buy_execution(weight),
					Order::DepositAsset {
						assets: vec![All],
						dest: Parachain(other_para_id).into(),
					},
				],
			},
			weight,
		), Outcome::Complete(w) if w == weight);
```

Reply:
I actually like to have a separate `r` variable here, it's much clearer to me than to parse through a `matches!` macro call.